runtime/native/monitor.cpp

```diff
@@ -27,11 +45,20 @@ Monitor &Monitor::Instance() {
 }
 
 void Monitor::enter(jobject obj) {
-
     {
         ScopedLock lock(&mutex);
-        // sanity check
-        Monitor::syncObjs.push_back(obj);
+
+        if (obj == nullptr) {
```

Comment:
We should just use a global reference to the `classLoadFakeObj` and not special case this here.
So when you generate the class loading code, instead of using a constant null it should be a reference to `classLoadFakeObj` which should be declared as a static global variable.

Reply:
That's a great idea. I just implemented this in the latest commit.